integrationtests/agent/adoption_test.go

```diff
@@ -0,0 +1,387 @@
+			createConfigMap(map[string]string{"foo": "bar"}, nil, nil)
+			createBundleDeployment("adopt-clean", true)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				if err := isConfigMapAdopted(&cm); err != nil {
+					return err
+				}
+				return mapPartialMatch(cm.Data, map[string]string{"foo": "bar", "key": "value"})
+			})
+		})
+	})
+
+	When("a bundle deployment adopts a resource with wrangler metadata", Label("wrangler-metadata"), func() {
+		It("verifies that the ConfigMap is adopted, its content merged and ownership changed", func() {
+			const (
+				objectSetHashKey   = "objectset.rio.cattle.io/hash"
+				objectSetIDKey     = "objectset.rio.cattle.io/id"
+				objectSetHashValue = "33ed67317c57ea78702e369c4c025f8df88553cc"
+				objectSetIDValue   = "some-assumed-old-id"
+			)
+			createConfigMap(
+				map[string]string{"foo": "bar"},
+				map[string]string{objectSetHashKey: objectSetHashValue},
+				map[string]string{objectSetIDKey: objectSetIDValue},
+			)
+			createBundleDeployment("adopt-wrangler-metadata", true)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				if err := isConfigMapAdopted(&cm); err != nil {
+					return err
+				}
+				if err := mapPartialMatch(cm.Data, map[string]string{"foo": "bar", "key": "value"}); err != nil {
+					return err
+				}
+				if mapPartialMatch(cm.Annotations, map[string]string{objectSetIDKey: objectSetIDValue}) == nil {
+					return fmt.Errorf("ObjectSet ID should have been updated")
+				}
+				if mapPartialMatch(cm.Labels, map[string]string{objectSetHashKey: objectSetHashValue}) == nil {
+					return fmt.Errorf("ObjectSet Hash should have been updated")
+				}
+				return nil
+			})
+		})
+	})
+
+	When("a bundle deployment adopts a resource with invalid wrangler metadata", Label("wrangler-metadata"), func() {
+		It("verifies that the ConfigMap is adopted and its content merged", func() {
+			createConfigMap(
+				map[string]string{"foo": "bar"},
+				map[string]string{"objectset.rio.cattle.io/hash": "234"},
+				map[string]string{"objectset.rio.cattle.io/id": "$#@"},
+			)
+			createBundleDeployment("adopt-invalid-wrangler-metadata", true)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				if err := isConfigMapAdopted(&cm); err != nil {
+					return err
+				}
+				return mapPartialMatch(cm.Data, map[string]string{"foo": "bar", "key": "value"})
+			})
+		})
+	})
+
+	When("a bundle deployment adopts a resource with random metadata", Label("random-metadata"), func() {
+		It("verifies that the ConfigMap is adopted and its content merged", func() {
+			createConfigMap(
+				map[string]string{"foo": "bar"},
+				map[string]string{"foo": "234"},
+				map[string]string{"bar": "xzy"},
+			)
+			createBundleDeployment("adopt-random-metadata", true)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				if err := isConfigMapAdopted(&cm); err != nil {
+					return err
+				}
+				return mapPartialMatch(cm.Data, map[string]string{"foo": "bar", "key": "value"})
```

Comment:
Same question as in the previous test case: do we want to validate that the random label and annotations are _not_ included in `cm`?

Reply:
I added that. The result is interesting. As resources deployed from bundle deployments are added the hash as label and the ID as annotation, we do not expect those to be preserved. But all other labels and annotations should be preserved.